raiden-ts/src/transport/epics.ts

```diff
@@ -1503,3 +1520,352 @@ export const deliveredEpic = (
+      filter(
+        (event) =>
+          event.getType() === ('m.call.candidates' as EventType) &&
+          event.getSender() === peerId &&
+          event.event?.content?.call_id === callId,
+      ),
+      tap((e) => log.debug('RTC: received candidates', callId, e.event?.content?.candidates)),
+      mergeMap((event) => from<RTCIceCandidateInit[]>(event.event?.content?.candidates ?? [])),
+      mergeMap((candidate) =>
+        defer(() => connection.addIceCandidate(new RTCIceCandidate(candidate))).pipe(
+          catchError((err) => {
+            log.error('RTC: error setting candidate, ignoring', err);
+            return EMPTY;
+          }),
+        ),
+      ),
+    ),
+  ).pipe(ignoreElements());
+}
+
+// setup channel for caller
+function setupCallerChannel$(
+  matrix: MatrixClient,
+  start$: Subject<null>,
+  info: CallInfo,
+  { httpTimeout }: RaidenConfig,
+  deps: Pick<RaidenEpicDeps, 'log' | 'latest$' | 'config$'>,
+) {
+  const { callId, peerId, peerAddress } = info;
+  const { log, latest$, config$ } = deps;
+
+  const connection = new RTCPeerConnection({
+    iceServers: [{ urls: 'stun:stun.l.google.com:19302' }],
+  });
+  // we relay on retries, no need to enforce ordered
+  const channel = connection.createDataChannel(callId, { ordered: false });
+  return merge(
+    // despite 'never' emitting, candidates$ have side-effects while/when subscribed
+    handleCandidates$(connection, matrix, start$, info, deps),
+    defer(() => connection.createOffer()).pipe(
+      mergeMap((offer) => {
+        connection.setLocalDescription(offer);
+        const content = {
+          call_id: callId,
+          lifetime: httpTimeout,
+          version: 0,
+          offer,
+        };
+        return merge(
+          // wait for answer
+          fromEvent<MatrixEvent>(matrix, 'event').pipe(
+            filter<MatrixEvent>(
+              // FIXME: remove any when MatrixEvent type exposes getAge & getContent methods
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+              (event: any) =>
+                event.getType() === ('m.call.answer' as EventType) &&
+                event.getSender() === peerId &&
+                event.getContent()?.call_id === callId &&
+                event.getAge() <= (event.getContent()?.lifetime ?? httpTimeout) &&
+                !!event.getContent()?.answer,
+            ),
+          ),
+          // send invite with offer
+          waitMemberAndSend$(peerAddress, matrix, 'm.call.invite' as EventType, content, {
+            log,
+            latest$,
+            config$,
+          }).pipe(
+            tap((e) => log.debug('RTC: sent invite', callId, e)),
+            ignoreElements(),
+          ),
+        );
+      }),
+      take(1),
+      tap(() => log.info('RTC: got answer', callId)),
+      map((event) => {
+        connection.setRemoteDescription(new RTCSessionDescription(event.event.content.answer));
```

Comment:
What is the description used for?

Reply:
These description are part of WebRTC protocol, this is how you set the exchanged RTC `offer`/`answer`s, by setting your side with `setLocalDescription` and the other side with `setRemoteDescription`; the exchange itself is done through signaling, in our case, dedicated matrix events.

Reply:
Thank you 🙏